### PR TITLE
Feature/eslint

### PR DESCRIPTION
**Pull request by Le-Caignec**
It seems that with node 14 the eslint raised and issue in dapp repo but not with node 18. I didn't find why.

### CODE REVIEW

File: dapp/package.json [hunk 1]
```diff
@@ -13,7 +13,8 @@
     "test": "NODE_OPTIONS=--experimental-vm-modules jest",
     "ctest": "NODE_OPTIONS=--experimental-vm-modules jest --silent=true --coverage",
     "format": "prettier --write \"src/**/*.js\" \"tests/**/*.js\"",
-    "check-format": "prettier --check \"src/**/*.js\" \"tests/**/*.js\""
+    "check-format": "prettier --check \"src/**/*.js\" \"tests/**/*.js\"",
+    "lint": "eslint . --ext .js"
```
Comment (PierreJeanjacquot) [inline dapp/package.json:17]: useless `--ext`

```suggestion
    "lint": "eslint ."
```

File: deployment-dapp/package.json [hunk 2]
```diff
@@ -9,7 +9,8 @@
     "push-dapp-secret": "ts-node --esm ./src/pushSecretScript.ts",
     "publish-sell-order": "ts-node --esm ./src/publishSellOrderScript.ts",
     "revoke-sell-order": "ts-node --esm ./src/revokeSellOrderScript.ts",
-    "configure-ens": "ts-node --esm ./src/configureEnsNameScript.ts"
+    "configure-ens": "ts-node --esm ./src/configureEnsNameScript.ts",
+    "lint": "eslint . --ext .ts"
```
Comment (PierreJeanjacquot) [inline deployment-dapp/package.json:13]: useless  `--ext`
```suggestion
    "lint": "eslint ."
```

File: deployment-dapp/.eslintrc.json [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    "prettier",
+    "plugin:sonarjs/recommended"
+  ],
+  "rules": {
+    "import/no-extraneous-dependencies": [
+      "error",
+      {
+        "devDependencies": true
+      }
+    ],
```
Comment (PierreJeanjacquot) [inline deployment-dapp/.eslintrc.json:22]: we don't need this, every dependency is listed in the dependencies

```suggestion
```

File: dapp/.eslintrc.json [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+      "error",
+      {
+        "devDependencies": true
+      }
+    ],
+    "sonarjs/cognitive-complexity": "warn"
+  },
+  "overrides": [
+    {
+      "files": ["./tests/**/*.ts"],
```
Comment (PierreJeanjacquot) [inline dapp/.eslintrc.json:25]: here are only js files
```suggestion
      "files": ["./tests/**/*.js"],
```

File: dapp/.eslintrc.json [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    "plugin:sonarjs/recommended",
+    "plugin:jest/recommended"
+  ],
+  "rules": {
+    "import/no-extraneous-dependencies": [
+      "error",
+      {
+        "devDependencies": true
+      }
+    ],
```
Comment (PierreJeanjacquot) [inline dapp/.eslintrc.json:20]: move `import/no-extraneous-dependencies` config in overrides for tests

File: .eslintrc.json [hunk 6]
```diff
@@ -1,18 +1,35 @@
...
+    "plugin:sonarjs/recommended",
+    "plugin:jest/recommended"
+  ],
+  "rules": {
+    "import/no-extraneous-dependencies": [
+      "error",
+      {
+        "devDependencies": true
+      }
+    ],
```
Comment (PierreJeanjacquot) [inline .eslintrc.json:22]: move `import/no-extraneous-dependencies` config in overrides for tests

**Review by PierreJeanjacquot — CHANGES_REQUESTED** [review]
see comments

File: dapp/package.json (hunk 1 above)
Comment (Le-Caignec) [reply dapp/package.json]: Commit: db5437f460817093eba89f2508f8dbed68b731f9

File: dapp/.eslintrc.json (hunk 5 above)
Comment (Le-Caignec) [reply dapp/.eslintrc.json]: Commit: 5357a4118273f2f1acedd9ffe677b84de3b4e65e

File: .eslintrc.json (hunk 6 above)
Comment (Le-Caignec) [reply .eslintrc.json]: Commit: d6b6fd2d410037788fa3246d57d9cea137972b22

**Review by PierreJeanjacquot — APPROVED** [review]
LGTM

**Pull request merged**